extmod/uasyncio/event.py

```diff
@@ -29,3 +33,24 @@ async def wait(self):
             core.cur_task.data = self.waiting
             yield
         return True
+
+
+_MP_STREAM_POLL = const(3)
+
+# MicroPython-extension: Event class that can be set from IRQs or scheduler context.
+# Implementation is a stream that asyncio will poll until a flag is set.
+class ThreadSafeEvent(io.IOBase):
```

Comment:
hmm... this now puts a requirement on all ports to have `uio.IOBase`... is it worth guarding this code in `hasattr(uio, 'IOBase')`?